sinks/cortex/cortex.go

```diff
@@ -207,66 +208,32 @@ func (s *CortexMetricSink) Flush(ctx context.Context, metrics []samplers.InterMe
 		return sinks.MetricFlushResult{}, nil
 	}
 
-	if s.batchWriteSize == 0 || len(metrics) <= s.batchWriteSize {
-		err := s.writeMetrics(ctx, metrics)
-		if err == nil {
-			flushedMetrics = len(metrics)
-		} else {
-			s.logger.Error(err)
-			droppedMetrics = len(metrics)
-		}
-
-		return sinks.MetricFlushResult{MetricsFlushed: flushedMetrics, MetricsDropped: droppedMetrics}, err
+	var allErrs *multierror.Error
+	batchSize := s.batchWriteSize
+	if s.batchWriteSize == 0 {
+		batchSize = len(metrics)
 	}
-
-	doIfNotDone := func(fn func() error) error {
+batching:
+	for i := 0; i < len(metrics); i += batchSize {
+		end := i + batchSize
+		if end > len(metrics) {
+			end = len(metrics)
+		}
+		batch := metrics[i:end]
 		select {
 		case <-ctx.Done():
-			return errors.New("context finished before completing metrics flush")
+			droppedMetrics += len(metrics[i:])
+			break batching
 		default:
-			return fn()
-		}
-	}
-
-	var batch []samplers.InterMetric
-	for _, metric := range metrics {
-		err := doIfNotDone(func() error {
-			batch = append(batch, metric)
-			if len(batch)%s.batchWriteSize == 0 {
-				err := s.writeMetrics(ctx, batch)
-				if err != nil {
-					return err
-				}
-
-				flushedMetrics += len(batch)
-				batch = []samplers.InterMetric{}
+			err := s.writeMetrics(ctx, batch)
+			if err != nil {
+				allErrs = multierror.Append(allErrs, err)
```

Comment:
If I'm reading this right...
Previously if we had a single failure, we stopped processing (it returned from the method). Now it will continue through the batch?

This sounds like a fix, but might cause unintended build up of failures on remote failures.

I think, if this was intended, a test should be added for it which failed before and passes now. That'll help show @arnavdugar-stripe the functional change as well for validation.